flink-test-utils-parent/flink-test-utils-junit/src/test/java/org/apache/flink/testutils/logging/TestLoggerResourceTest.java

```diff
@@ -0,0 +1,119 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.testutils.logging;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.Test;
+import org.slf4j.event.Level;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * {@link TestLoggerResourceTest} ensures that the use of {@link TestLoggerResource} combined with
+ * other loggers (including multiple instances of itself) does not lead to unexpected behavior.
+ */
+public class TestLoggerResourceTest {
+    final String parentLoggerName = TestLoggerResourceTest.class.getName() + ".parent";
+    final String childLoggerName = parentLoggerName + ".child";
+    final Logger parentLogger = LogManager.getLogger(parentLoggerName);
+    final Logger childLogger = LogManager.getLogger(childLoggerName);
+
+    @Test
+    public void loggerWithoutChild() throws Throwable {
+        try (TestLoggerResource.SingleTestResource parentResource =
+                TestLoggerResource.asSingleTestResource(parentLoggerName, Level.INFO)) {
+            parentLogger.info("child-info");
+            parentLogger.debug("child-debug");
+            List<String> msgs = parentResource.getMessages();
+            assertThat(msgs).containsExactly("child-info");
+        }
+    }
+
+    @Test
+    public void loggerIsAlreadyDefined() throws Throwable {
+        // Sometimes a logger is already defined for the same class (e.g., to debug the test on
+        // failure.
+        try (TestLoggerResource.SingleTestResource outerResource =
+                        TestLoggerResource.asSingleTestResource(parentLoggerName, Level.INFO);
+                TestLoggerResource.SingleTestResource innerResource =
+                        TestLoggerResource.asSingleTestResource(parentLoggerName, Level.DEBUG)) {
+            parentLogger.info("child-info");
+            parentLogger.debug("child-debug");
+            List<String> msgsInner = innerResource.getMessages();
+            assertThat(msgsInner).containsExactly("child-info", "child-debug");
+            List<String> msgsOuter = outerResource.getMessages();
+            assertThat(msgsOuter).contains("child-info");
```

Comment:
```suggestion
            assertThat(msgsOuter).containsExactly("child-info");
```
?

Reply:
The reason why the outerResource doesn't contain child-debug, despite using the same loggerName, is because the previous appender that defined a filter is re-applied to the new logger, yes?

Reply:
ok